components/frontend/src/source/SourceEntityAttribute.js

```diff
@@ -5,10 +5,12 @@ import { TimeAgoWithDate } from '../widgets/TimeAgoWithDate';
 
 export function SourceEntityAttribute({ entity, entity_attribute }) {
     let cell_contents = entity[entity_attribute.key] ?? "";
-    cell_contents = cell_contents && entity_attribute.type === "datetime" ? <TimeAgoWithDate dateFirst date={cell_contents} /> : cell_contents;
-    cell_contents = cell_contents && entity_attribute.type === "date" ? <TimeAgoWithDate dateFirst noTime date={cell_contents} /> : cell_contents;
-    cell_contents = cell_contents && entity_attribute.type === "integer" ? Math.round(cell_contents) : cell_contents;
-    cell_contents = cell_contents && entity_attribute.type === "status" ? <StatusIcon status={cell_contents} /> : cell_contents;
+    const type = entity_attribute.type ?? "";
+    cell_contents = cell_contents && type === "datetime" ? <TimeAgoWithDate dateFirst date={cell_contents} /> : cell_contents;
+    cell_contents = cell_contents && type === "date" ? <TimeAgoWithDate dateFirst noTime date={cell_contents} /> : cell_contents;
+    cell_contents = cell_contents && type.includes("integer") ? Math.round(cell_contents).toString() : cell_contents;
+    cell_contents = cell_contents && type.includes("percentage") ? cell_contents + "%" : cell_contents;
```

Comment:
Do lines 11 and 12 indeed both get triggered for type "integer_percentage" ? I'm not familiar with how the `&&` operator makes the object interact with itself here

Reply:
Not sure what "interact with itself" means? Anyway, `&&` has a higher priority than `x ? y : z`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence